sdv/constraints/tabular.py

```diff
@@ -723,7 +746,11 @@ def reverse_transform(self, table_data):
         data = data * (high - low) + low
         data = data.clip(low, high)
 
-        table_data[self.constraint_column] = data
+        if self._is_datetime:
+            table_data[self.constraint_column] = pd.to_datetime(data)
```

Comment:
@csala I know we talked about storing the original column's dtype to use in the `to_datetime` conversion here, but I don't think that's necessary since pandas seems to only have `datetime[ns]`(see [doc](https://pandas.pydata.org/pandas-docs/stable/user_guide/basics.html?highlight=basics#dtypes)). If you do something like `pd.Series([pd.to_datetime(1627497875, unit='s')])` it will give you back a data series of `dtype: datetime64[ns]`